File: nbrk_currency_rates/src/main/java/com/nbrk/rates/db/DbOpenHelper.java
package com.nbrk.rates.db;

import android.content.Context;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteOpenHelper;
import com.nbrk.rates.model.CurrencyRatesItem;

import static nl.qbusict.cupboard.CupboardFactory.cupboard;

public class DbOpenHelper extends SQLiteOpenHelper {

    private static final String DATABASE_NAME = "currency_rates.db";
    private static final int DATABASE_VERSION = 4;

    static {
        // register our models
        cupboard().register(CurrencyRatesItem.class);
    }

    public DbOpenHelper(Context context) {
        super(context, DATABASE_NAME, null, DATABASE_VERSION);
    }

    @Override
    public void onCreate(SQLiteDatabase db) {
        // this will ensure that all tables are created
        cupboard().withDatabase(db).createTables();
        // add indexes and other database tweaks
    }

    @Override
    public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
        // this will upgrade tables, adding columns and new tables.
        // Note that existing columns will not be converted
        cupboard().withDatabase(db).upgradeTables();
        // do migration work
    }
}

File: nbrk_currency_rates/src/main/kotlin/com/nbrk/rates/data/db/DbOpenHelper.kt
package com.nbrk.rates.data.db

import android.content.Context
import android.database.sqlite.SQLiteDatabase
import com.nbrk.rates.App
import com.nbrk.rates.data.db.tables.CurrencyRates
import org.jetbrains.anko.db.*

class DbOpenHelper(context: Context = App.instance) :
  ManagedSQLiteOpenHelper(context, DbOpenHelper.DATABASE_NAME, null, DbOpenHelper.DATABASE_VERSION) {

  companion object {
    private val DATABASE_NAME = "currency_rates.db"
    private val DATABASE_VERSION = 5
    val instance: DbOpenHelper by lazy { DbOpenHelper() }
  }

  override fun onCreate(db: SQLiteDatabase) {
    db.createTable(CurrencyRates.TABLE_NAME, true,
      CurrencyRates.COL_DATE to INTEGER,
      CurrencyRates.COL_CURRENCY_CODE to TEXT,
      CurrencyRates.COL_CURRENCY_NAME to TEXT,
      CurrencyRates.COL_QUANTITY to INTEGER,
      CurrencyRates.COL_PRICE to REAL,
      CurrencyRates.COL_INDEX to TEXT,
      CurrencyRates.COL_CHANGE to REAL
    )
  }

  override fun onUpgrade(db: SQLiteDatabase, oldVersion: Int, newVersion: Int) {
    db.dropTable(CurrencyRates.TABLE_NAME, true)
    db.dropTable("CurrencyRatesItem", true)
    onCreate(db)
  }
}
